server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/transaction/TransactionController.java

```diff
@@ -574,6 +575,10 @@ private ModelAndView getSparqlUpdateResult(Transaction transaction, HttpServletR
 			sparqlUpdateString = request.getParameter(Protocol.UPDATE_PARAM_NAME);
 		}
 
+		if (null == sparqlUpdateString) {
+			throw new ClientHTTPException(SC_NOT_ACCEPTABLE, "Could not read SPARQL update string from body.");
```

Comment:
I am on the fence over whether this should be a 404 instead of a 406, but I can see the reasoning.

Reply:
I was really uncertain exactly how to present it to the end user as well. It's a very particular corner-case. It's a 400/404, with a root cause of 406.